components/merkledb/src/proof_map_index/mod.rs

```diff
@@ -202,11 +209,20 @@ impl BinaryAttribute for Option<ProofPath> {
     }
 }
 
-impl<T, K, V> ProofMapIndex<T, K, V>
+/// Hashed variant of the `ProofMapIndex`, useful for arbitrary keys that implement
+/// `ObjectHash` trait.
+pub type HashedProofMap<T, K, V> = ProofMapIndex<T, K, V, Hashed>;
+
+/// Raw variant of the `ProofMapIndex`, useful for keys that mapped directly to
+/// `ProofPath`. For example `Hash` and `PublicKey`.
+pub type RawProofMap<T, K, V> = ProofMapIndex<T, K, V, Raw>;
```

Comment:
I don't like `RawProofMap` name. "Raw" is usually treated like something low-level, but here it means only the key hashing approach.

IMHO, it'll be better to call current `ProofMapIndex` a `RawProofMapIndex` (since it requires us to provide a `Style` parameter), and this one (as covering most of the cases) to rename to the `ProofMapIndex`.

As a result, for situations where key should be hashed, a `ProofMapIndex` will be used (which is consistent and familiar), and for situations where key is already "hashed" it will become a `HashedProofMapIndex`.

Reply:
Agree, but my variant is slightly different. I've changed 
```
ProofMapIndex -> ProofMapIndexBase
HashedProofMap -> ProofMapIndex
RawProofMap -> RawProofMapIndex
```

Reply:
Yeah, that's OK.